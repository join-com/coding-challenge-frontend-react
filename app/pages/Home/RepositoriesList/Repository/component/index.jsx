import React from 'react';
import PropTypes from 'prop-types';
import { FormattedNumber } from 'react-intl';

import IssueIcon from '../IssueIcon';
import IssuesLink from '../IssuesLink';
import RepositoryLink from '../RepositoryLink';

import styles from './styles.scss';

export const Repository = ({currentUser, repository}) => (
  <div className={styles.wrapper}>
    <RepositoryLink href={repository.html_url} target="_blank">
      {repository.owner.login === currentUser ? repository.name : repository.full_name}
    </RepositoryLink>
    <IssuesLink href={`${repository.html_url}/issues`} target="_blank">
      <IssueIcon className={styles.icon} />
      <FormattedNumber value={repository.open_issues_count} />
    </IssuesLink>
  </div>
);

Repository.propTypes = {
  repository: PropTypes.object,
  currentUser: PropTypes.string,
};

export default Repository;
