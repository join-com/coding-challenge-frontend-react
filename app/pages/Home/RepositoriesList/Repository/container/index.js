import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { makeSelectSelectedUsername } from '../../../../../store/github/selectors';

import Repository from '../component';

export default connect(createStructuredSelector({
  currentUser: makeSelectSelectedUsername(),
}))(Repository);
