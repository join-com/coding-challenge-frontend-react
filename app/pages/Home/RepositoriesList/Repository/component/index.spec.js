import React from 'react';
import renderer from 'react-test-renderer';

import { IntlProvider } from 'react-intl';

import Repository from './';

const currentUser = 'alexander-elgin';

const currentUserRepository = {
  id: 52143857,
  name: 'ta-maxlength',
  full_name: 'alexander-elgin/ta-maxlength',
  owner: {
    login: 'alexander-elgin',
  },
  html_url: 'https://github.com/alexander-elgin/ta-maxlength',
  open_issues_count: 3,
};

const anotherUserRepository = {
  id: 33667892,
  name: 'meetermaid',
  full_name: 'rasp3175/meetermaid',
  owner: {
    login: 'rasp3175',
  },
  html_url: 'https://github.com/rasp3175/meetermaid',
  open_issues_count: 0,
};

describe('<Repository />', () => {
  it('hides the repository owner name if it matches the current user name', () => {
    expect(renderer.create(
      <IntlProvider locale="en">
        <Repository currentUser={currentUser} repository={currentUserRepository} />
      </IntlProvider>
    ).toJSON()).toMatchSnapshot();
  });

  it('shows the repository owner name if it does not match the current user name', () => {
    expect(renderer.create(
      <IntlProvider locale="en">
        <Repository currentUser={currentUser} repository={anotherUserRepository} />
      </IntlProvider>
    ).toJSON()).toMatchSnapshot();
  });
});
