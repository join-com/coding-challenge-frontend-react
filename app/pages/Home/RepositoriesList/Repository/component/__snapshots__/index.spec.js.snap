// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repository /> hides the repository owner name if it matches the current user name 1`] = `
<div
  className="wrapper"
>
  <a
    className="component__A-sc-1qxsdu7-0 component__RepoLink-sc-2iw2dp-0 cbLOVz"
    href="https://github.com/alexander-elgin/ta-maxlength"
    target="_blank"
  >
    ta-maxlength
  </a>
  <a
    className="component__A-sc-1qxsdu7-0 component__IssuesLink-sc-1rshot1-0 ezZxTa"
    href="https://github.com/alexander-elgin/ta-maxlength/issues"
    target="_blank"
  >
    <svg
      className="icon"
      height="1em"
      width="0.875em"
    >
      <path
        d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7S10.14 13.7 7 13.7 1.3 11.14 1.3 8s2.56-5.7 5.7-5.7m0-1.3C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7S10.86 1 7 1z m1 3H6v5h2V4z m0 6H6v2h2V10z"
      />
    </svg>
    <span>
      3
    </span>
  </a>
</div>
`;

exports[`<Repository /> shows the repository owner name if it does not match the current user name 1`] = `
<div
  className="wrapper"
>
  <a
    className="component__A-sc-1qxsdu7-0 component__RepoLink-sc-2iw2dp-0 cbLOVz"
    href="https://github.com/rasp3175/meetermaid"
    target="_blank"
  >
    rasp3175/meetermaid
  </a>
  <a
    className="component__A-sc-1qxsdu7-0 component__IssuesLink-sc-1rshot1-0 ezZxTa"
    href="https://github.com/rasp3175/meetermaid/issues"
    target="_blank"
  >
    <svg
      className="icon"
      height="1em"
      width="0.875em"
    >
      <path
        d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7S10.14 13.7 7 13.7 1.3 11.14 1.3 8s2.56-5.7 5.7-5.7m0-1.3C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7S10.86 1 7 1z m1 3H6v5h2V4z m0 6H6v2h2V10z"
      />
    </svg>
    <span>
      0
    </span>
  </a>
</div>
`;
